service/internal/subjectmappingbuiltin/subject_mapping_builtin_actions.go

```diff
@@ -0,0 +1,87 @@
+package subjectmappingbuiltin
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/opentdf/platform/lib/flattening"
+	entityresolutionV2 "github.com/opentdf/platform/protocol/go/entityresolution/v2"
+	"github.com/opentdf/platform/protocol/go/policy"
+	"github.com/opentdf/platform/protocol/go/policy/attributes"
+)
+
+type AttributeValueFQNsToActions map[string][]*policy.Action
+
+type EntityIDsToEntitlements map[string]AttributeValueFQNsToActions
+
+func EvaluateSubjectMappingMultipleEntitiesWithActions(
+	attributeMappings map[string]*attributes.GetAttributeValuesByFqnsResponse_AttributeAndValue,
+	entityRepresentations []*entityresolutionV2.EntityRepresentation,
+) (EntityIDsToEntitlements, error) {
+	results := make(map[string]AttributeValueFQNsToActions)
```

Comment:
```suggestion
	results := make(map[string]AttributeValueFQNsToActions, len(entityRepresentations))
```

minor optimization if you're interested